stdlib/2/__builtin__.pyi

```diff
@@ -1370,20 +1370,34 @@ else:
     # This is only available after from __future__ import print_function.
     def print(*values: object, sep: Optional[Text] = ..., end: Optional[Text] = ..., file: Optional[_Writer] = ...) -> None: ...
 
+_E = TypeVar("_E", contravariant=True)
```

Comment:
I'm having a hard time understanding the need for contravariance here. I tested with the following file:

```
class A:
    pass


class B(A):
    pass


class C(B):
    pass


class PowPow:
    def __pow__(self, exp: B) -> str:
        return ""


reveal_type(pow(PowPow(), A()))
reveal_type(pow(PowPow(), B()))
reveal_type(pow(PowPow(), C()))
```

But mypy produces the same output regardless of whether I make `_E` contravariant, covariant, or invariant:
```
powpow.py:18: error: Cannot infer type argument 1 of "pow"
powpow.py:18: note: Revealed type is 'Any'
powpow.py:19: note: Revealed type is 'builtins.str*'
powpow.py:20: note: Revealed type is 'builtins.str*'
```

Reply:
I wasn't sure about the variance. Looks like your're right. Is my `_T_co` needed for the return value?

Reply:
`test/mypy_test.py` reports an error when `_E` and `_M` are not contravariant:
```
$ python3 tests/mypy_test.py
running mypy --python-version 3.9 --strict-optional --no-site-packages --show-traceback --no-implicit-optional --disallow-any-generics --disallow-subclassing-any # with 970 files
stdlib/2and3/builtins.pyi:1375: error: Invariant type variable '_E' used in protocol where contravariant one is expected
stdlib/2and3/builtins.pyi:1377: error: Invariant type variable '_E' used in protocol where contravariant one is expected
stdlib/2and3/builtins.pyi:1377: error: Invariant type variable '_M' used in protocol where contravariant one is expected
```

Reply:
OK, I guess we should believe mypy here.

Reply:
mypy's a little strict about variance in generic protocols. Say you subclassed PowPow with PowPow2 that accepted a wider variety of types than just B's. PowPow2 does obey LSP, and is a nominal subtype of PowPow, but is not a subtype of Pow[B] if the type variables are invariant, resulting in weird intransitivity.